elliottlib/cli/__main__.py

```diff
@@ -200,12 +208,37 @@ def find_bugs(runtime, advisory, default_advisory_type, mode, status, id, from_d
                                                                                     bug.summary[:60]))
 
     if len(flag) > 0:
-        for bug in bug_ids:
+        for bug in bugs:
             for f in flag:
                 bug.updateflags({f: "+"})
 
-    if advisory is not False:
-        elliottlib.errata.add_bugs_with_retry(advisory, [bug.id for bug in bug_ids], False)
+    if advisory is not False and not default_advisory_type:  # `--add ADVISORY_NUMBER` should respect the user's wish and attach all available bugs to whatever advisory is specified.
+        elliottlib.errata.add_bugs_with_retry(advisory, [bug.id for bug in bugs], False)
+        return
+
+    # if --use-default-advisory or --into-default-advisories is given, we need to determine which bugs should be swept into which advisory
+    if not into_default_advisories and not default_advisory_type:
+        return
+    impetus_bugs = {}  # key is rpm impetus ("rpm", "image", "extras"), value is a set of bug IDs.
+    m = re.match(r"rhaos-(\d+).(\d+)", runtime.branch)  # extract OpenShift version from the branch name. there should be a better way...
+    if not m:
+        raise ElliottFatalError(f"Unable to determine OpenShift version from branch name {runtime.branch}.")
+    # @lmeyer: simple and stupid would still be keeping the logic in python, possibly with config flags for branched logic. until that logic becomes too ugly to keep in python, i suppose..
+    major_version = int(m[1])
+    if major_version < 4:  # for 3.x, all bugs should go to the rpm advisory
+        impetus_bugs["rpm"] = {bug.id for bug in bugs}
+    else:  # for 4.x, optional operators bugs should be swept to the "extras" advisory, while other bugs should be swept to "image" advisory.
+        # a way to identify operator-related bugs is by its "Component" value. temporarily hardcode here until we need to move it to ocp-build-data.
```

Comment:
you can look in `oc adm release info --commits "${RELEASE_PULLSPEC}"` to get a list of repos and commits, and then you can look into their `OWNERS` to get a Bugzilla component.  For example:

```console
$ oc adm release info -o json quay.io/openshift-release-dev/ocp-release:4.5.2-x86_64 | jq -r '.references.spec.tags[].annotations | .["io.openshift.build.source-location"] as $repo | ($repo | sub("https://github.com/"; "")) as $org_repo | .["io.openshift.build.commit.id"] as $commit | select($commit | length > 0) | "https://raw.githubusercontent.com/" + $org_repo + "/" + $commit + "/OWNERS"' | sort | uniq | while read URI; do COMPONENT="$(curl -s "${URI}" | yaml2json | jq -r .component)"; echo "${URI} -> ${COMPONENT}"; done
https://raw.githubusercontent.com/openshift/aws-pod-identity-webhook/2251f7e2fc9194aa3a57cd9dc1de4a71e9367df5/OWNERS -> Cloud Credential Operator
https://raw.githubusercontent.com/openshift/baremetal-operator/e276e9f64e3c49abbd13267914bfa7fef1fa3a9b/OWNERS -> Bare Metal Hardware Provisioning
https://raw.githubusercontent.com/openshift/baremetal-runtimecfg/855ed6b0022c8c9557138e58514bb4301f4043f5/OWNERS -> null
https://raw.githubusercontent.com/openshift/builder/bb6e41a1e23a61e070274f778d86d4211bdb41ff/OWNERS -> null
https://raw.githubusercontent.com/openshift/cloud-credential-operator/863098a4b783c2194dac62eddfc55fc44c8f7ae4/OWNERS -> Cloud Credential Operator
...
```

That's not going to be perfect, but if we fill in some remaining `component` values where that gives `null` today, it should be pretty close for identifying bugs which affect a release image directly.  Things like machine-os-content that pull in content via RPMs would be more work.  Everything else would be `extras`, right?

Reply:
if the OWNERS file specifies BZ info then it's reflected in labels on each image built from it. but that's not terribly useful here, as we aren't looking at builds here but rather bugs, and trying to determine if they pertain to the payload or to extras.

we don't yet have a reliable link between the bug and the build it corresponds to. if we established that... it might be easier to build a mapping between build usage and related bugs.

Reply:
[`oc adn release info --bugs ...`][1] is heavy, but will list bugs addressed between two releases.  Except for the first GA in a z-stream, because during feature development we don't require formal bug -> PR links.

[1]:https://github.com/openshift/oc/blob/b6f8f97e76dd72ee9a7189701d548f16bd38de27/pkg/cli/admin/release/info.go#L93